### PR TITLE
Add DTLS sequence number to endpoint context

### PR DESCRIPTION
**Pull request by adamsero**

### CODE REVIEW

File: element-connector/src/main/java/org/eclipse/californium/elements/DtlsEndpointContext.java [hunk 1]
```diff
@@ -56,6 +56,12 @@ public class DtlsEndpointContext extends MapBasedEndpointContext {
 	 */
 	public static final Definition<Long> KEY_HANDSHAKE_TIMESTAMP = new Definition<>("DTLS_HANDSHAKE_TIMESTAMP",
 			Long.class, ATTRIBUTE_DEFINITIONS);
+	/**
+	 * The name of the attribute that contains the dtls packet sequence number
+	 * during the DTLS session as {@link Number}.
+	 */
+	public static final Definition<Long> DTLS_SEQUENCE_NUMBER = new Definition<>(
+			KEY_PREFIX_NONE_CRITICAL + "DTLS_SEQUENCE_NUMBER", Long.class, ATTRIBUTE_DEFINITIONS);
```
Comment (boaks) [inline element-connector/src/main/java/org/eclipse/californium/elements/DtlsEndpointContext.java:64]: As I wrote in my comment on issue #2085, there are two different sequence number in a session. One for reading and one for writing. I would prefer to reflect that by naming it `DTLS_READ_SEQUENCE_NUMBER`.

WDYT?

**Comment by boaks** [comment]
Beside of the name, LGTM.

File: element-connector/src/main/java/org/eclipse/californium/elements/DtlsEndpointContext.java (hunk 1 above)
Comment (adamsero) [reply element-connector/src/main/java/org/eclipse/californium/elements/DtlsEndpointContext.java]: Sorry, I focused on `KEY_PREFIX_NONE_CRITICAL` and completely overlooked that you also wanted to add `READ` in the name too, my bad 😅

Comment (boaks) [reply element-connector/src/main/java/org/eclipse/californium/elements/DtlsEndpointContext.java]: Thanks for your contribution! You're welcome.

I don't think, that the "DTLS_WRITE_SEQUENCE_NUMBER" will be added, but if, then it's easier to use a specific name.

**Pull request merged**